src/bin/add/args.rs

```diff
@@ -106,6 +109,22 @@ impl Args {
 
         Ok(vec![dependency])
     }
+
+    fn get_upgrade_prefix(&self) -> &'static str {
```

Comment:
I didn't forget about this but I have been wrestling with the fact that `map` takes ownership of the option.

I have now 

```
fn get_upgrade_prefix(&self) -> Option<&'static str> {
		let flag_upgrade = &self.flag_upgrade; 
		
		flag_upgrade.map(|flag|
			match flag.to_uppercase().as_ref() {
                "NONE" => Some("="),
                "PATCH" => Some("~"),
                "MINOR" => Some("^"),
                "ALL" => Some(">="),
                _ => {
                    println!("WARN: cannot understand upgrade option \"{}\", using default", flag);
                    None
                },
            }
        ).unwrap_or(None)
    }
```


with compiler error 

```
src/bin/add/args.rs:119:3: 119:15 error: cannot move out of borrowed content [E0507]
src/bin/add/args.rs:119 		flag_upgrade.map(|flag|
    
```

I have [this playground](https://is.gd/1cptcS) that works so I'm confused. I think the problem is with the `&self` but I'm not sure what to do about it.